File: ontrack-extension-github/src/main/java/net/nemerosa/ontrack/extension/github/property/GitHubProjectConfigurationProperty.java
package net.nemerosa.ontrack.extension.github.property;

import lombok.Data;
import net.nemerosa.ontrack.extension.github.model.GitHubEngineConfiguration;
import net.nemerosa.ontrack.model.support.ConfigurationProperty;

@Data
public class GitHubProjectConfigurationProperty implements ConfigurationProperty<GitHubEngineConfiguration> {

    /**
     * Link to the GitHub configuration
     */
    private final GitHubEngineConfiguration configuration;

    /**
     * Repository name
     */
    private final String repository;

    /**
     * Indexation interval
     */
    private final int indexationInterval;

    /**
     * ID to the {@link net.nemerosa.ontrack.extension.issues.model.IssueServiceConfiguration} associated
     * with this repository.
     */
    private final String issueServiceConfigurationIdentifier;

}

File: ontrack-extension-github/src/main/java/net/nemerosa/ontrack/extension/github/property/GitHubProjectConfigurationProperty.kt
package net.nemerosa.ontrack.extension.github.property

import net.nemerosa.ontrack.extension.github.model.GitHubEngineConfiguration
import net.nemerosa.ontrack.model.annotations.APIDescription
import net.nemerosa.ontrack.model.docs.DocumentationType
import net.nemerosa.ontrack.model.support.ConfigurationProperty

/**
 * @property configuration Link to the GitHub configuration
 * @property repository Repository name
 * @property indexationInterval Indexation interval
 * @property issueServiceConfigurationIdentifier ID to the [net.nemerosa.ontrack.extension.issues.model.IssueServiceConfiguration] associated
 * with this repository.
 */
class GitHubProjectConfigurationProperty(
    @DocumentationType("String", description = "Name of the configuration")
    override val configuration: GitHubEngineConfiguration,
    @APIDescription("GitHub repository, ie. org/name")
    val repository: String,
    @APIDescription("How often to index the repository, in minutes. Use 0 to disable indexation.")
    val indexationInterval: Int,
    @APIDescription("Identifier for the issue service")
    val issueServiceConfigurationIdentifier: String?
) : ConfigurationProperty<GitHubEngineConfiguration>

